### PR TITLE
modify the warning

### PR DESCRIPTION
**Pull request by ColinCollins**
修正 ts 1.10 之后的脚本警告问题。
稍后会附带的修正 ts 相关的文档

### CODE REVIEW

File: cocos2d/core/platform/preprocess-class.js [hunk 1]
```diff
@@ -226,11 +226,16 @@ exports.getFullFormOfProperty = function (options, propname_dev, classname_dev)
                         'Please use the complete form.\n' +
                         'For example, if property is Texture2D\'s url array, the previous definition is:\n' +
                         '    %s: [cc.Texture2D],\n' +
-                        'Now it should be changed to:\n' +
+                        'Now in JS it should be changed to:\n' +
```
Comment (pandamicro) [inline cocos2d/core/platform/preprocess-class.js]: If you use JavaScript, it should be changed to:

File: cocos2d/core/platform/preprocess-class.js [hunk 2]
```diff
@@ -226,11 +226,16 @@ exports.getFullFormOfProperty = function (options, propname_dev, classname_dev)
...
                         '    %s: {\n' +
                         '      type: cc.Texture2D, // use \'type:\' to define an array of Texture2D objects\n' +
                         '      default: []\n' +
                         '    },\n' +
+                        'Now in TS it should be changed to:\n' +
```
Comment (pandamicro) [inline cocos2d/core/platform/preprocess-class.js]: If you use TypeScript, it should be changed to:

File: cocos2d/core/platform/preprocess-class.js [hunk 3]
```diff
@@ -257,11 +262,16 @@ exports.getFullFormOfProperty = function (options, propname_dev, classname_dev)
                                 'Please use the complete form.\n' +
                                 'For example, if the type is Texture2D, the previous definition is:\n' +
                                 '    %s: cc.Texture2D,\n' +
-                                'Now it should be changed to:\n' +
+                                'Now in JS it should be changed to:\n' +
```
Comment (pandamicro) [inline cocos2d/core/platform/preprocess-class.js]: 同上

File: cocos2d/core/platform/CCClassDecorator.js [hunk 4]
```diff
@@ -228,7 +228,10 @@ function genProperty (ctor, properties, propName, options, desc, cache) {
                 isDefaultValueSpecified &&
                 defaultValue == null
             ) {
-                cc.warnID(3656, JS.getClassName(ctor), propName);
+                // avoid the excess warning when the ts decorator format is error
```
Comment (pandamicro) [inline cocos2d/core/platform/CCClassDecorator.js]: Avoid excessive warning when the ts decorator format is wrong

**Review by pandamicro — CHANGES_REQUESTED** [review]

**Review by jareguo — APPROVED** [review]

**Review by pandamicro — APPROVED** [review]

**Pull request merged**